src/__tests__/heatmaps.test.ts

```diff
@@ -89,6 +89,27 @@ describe('heatmaps', () => {
         expect(posthog.heatmaps!.getAndClearBuffer()).toBeDefined()
     })
 
+    it('should handle empty mouse moves', async () => {
+        posthog.heatmaps?.['_onMouseMove']?.(new Event('mousemove'))
+
+        jest.advanceTimersByTime(posthog.heatmaps!.flushIntervalMilliseconds + 1)
+
+        expect(beforeSendMock).toBeCalledTimes(1)
+        expect(beforeSendMock.mock.lastCall[0]).toMatchObject({
+            event: '$$heatmap',
+            properties: {
+                $heatmap_data: {
+                    'http://replaced/': [
+                        {
+                            target_fixed: false,
```

Comment:
so, does this mean that the mousemove event has no clientX or clientY?

if so we're actually better ignoring them completely since the data is invalid without a coordinate.

Reply:
Yes, it's a mouse move event without any clientX or clientY. According to [mozilla dev](https://developer.mozilla.org/en-US/docs/Web/API/Event/Event) everything is optional, so it is valid to send an event with any data.

Yeah ignoring is probably a good call!

Reply:
you're welcome to add that to this PR - if you're interested?

i can do as follow-up otherwise

Reply:
we're already dropping events with no x and y during  ingestion, so it's only a nice-to-have not to omit them (although good to not do work)

so only open question for me is if you'd like to handle that in this PR?

Reply:
I think I'd leave that to you, I'm not sure where you'd implement that filtering.